.golangci.yml

```diff
@@ -0,0 +1,51 @@
+run:
+  # timeout for analysis
+  deadline: 10m
+
+linters-settings:
+  govet:
+    # Don't report about shadowed variables
+    check-shadowing: false
+  gofmt:
+    # simplify code: gofmt with `-s` option, true by default
+    simplify: true
+
+linters:
+  enable-all: true
+  disable:
+    # Global variables are used in many places throughout the code base.
+    - gochecknoglobals
+
+    # Some lines are over 80 characters on purpose and we don't want to make them
+    # even longer by marking them as 'nolint'.
+    - lll
+
+    # We don't care (enough) about misaligned structs to lint that.
+    - maligned
```

Comment:
maybe some of these could be enabled?

Reply:
Yeah, why not. I enabled `malign` globally but only fixed one struct that is used often. For the others the readability/order of the struct would suffer too much for very little benefit.

Also skipping the `prealloc` check only for test code.